src/com/epam/javatrainings/annotations/processor/MaxAndMinAnnotationsProcessor.java

```diff
@@ -0,0 +1,19 @@
+package com.epam.javatrainings.annotations.processor;
+
+import com.epam.javatrainings.annotations.annotation.Max;
+import com.epam.javatrainings.annotations.annotation.Min;
+import com.epam.javatrainings.annotations.exception.WrongDiscountException;
+
+import java.lang.reflect.Field;
+
+public class MaxAndMinAnnotationsProcessor {
+    public <T> MaxAndMinAnnotationsProcessor(Field field, T t) throws WrongDiscountException, IllegalAccessException {
+        field.setAccessible(true);
+        Min min = field.getAnnotation(Min.class);
+        Max max = field.getAnnotation(Max.class);
+        Integer discountRate = (Integer) field.get(t);
```

Comment:
May produce ClassCastException if the @Max or @Min were declared on another type